modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +41,618 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
+
+=== Input
+
+[source,JSON,linenums]
+----
+{ "ref": "name", "name": "Data Weave" }
+----
+
+==== Source
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload[(payload.ref)]
+----
+
+==== Output
+
+[source,JSON,linenums]
+----
+"Data Weave"
+----
+
+
+=== Example 
+This example shows how to dynamically select a single value with namesapce. It uses the value on the "ref" property to select the value with that key.
+
+=== Input
+
+[source,xml,linenums]
+----
+<root ref="table">
+<f:table xmlns:f="https://www.w3schools.com/furniture">Manzana</f:table>
+    <h:table xmlns:h="http://www.w3.org/TR/html4/">Banana</h:table>    
+</root>
+----
+
+==== Source
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+ns h http://www.w3.org/TR/html4/
+---
+payload.root.h#"$(payload.root.@ref)"
+----
+
+==== Output
+
+[source,JSON,linenums]
+----
+"Banana"
+----
+
+
+[[single_value_selector]]
+== Single-Value Selector
+
+This selector matches the value of the first key-value pair where the key
+matches the given selector name. The selector can be applied to an Object or
+to an Array. On an Array, the selector applies to all Object values inside the
+array and ignores all values that are not Object values. If no key matches,
+the value `null` returns.
+
+=== Example
+
+This example shows how the selector works on a simple JSON object.
+
+==== Input
+
+The input payload is an object with the key "name" and the
+value "DataWeave".
+
+[source,JSON,linenums]
+----
+{ "name": "Data Weave" }
+----
+
+==== Source
+
+The DataWeave script uses `payload.name` to select the
+value of the object that has the key `"name"`.
+Note that `payload.name` and `payload."name"` are both valid
+ways to perform the selection.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+The script outputs the String value of the input object, "Data Weave".
+
+[source,JSON,linenums]
+----
+"Data Weave"
+----
+
+=== Example
+
+On an Array, the selector applies to every element.
+
+
+==== Input
+
+The input payload is an array that contains two objects with the same
+key, "name".
+
+[source,JSON,linenums]
+----
+[
+  {
+    "name": "Arg"
+  },
+  {
+    "name": "Japan"
+  }
+]
+----
+
+==== Source
+
+The DataWeave script uses `payload.name` to select the
+value of any objects in input payload that have the key, "name".
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.name
+----
+
+==== Output
+
+The script outputs an array that contains the values of the input objects
+with the key "name": "Arg" and "Japan".
+
+[source,json, linenums]
+----
+["Arg", "Japan"]
+----
+
+[[namespace_example1]]
+== Namespace Selector
+
+The DataWeave namespace selector supports the use of a name that matches a
+namespace or a local part. To match a namespace, the name must match the
```

Comment:
Does your audience know what "a local part" is? Because I don't...lol....

Reply:
there's an example. It's a common term.

Reply:
it's explained later also